src/MobileUI/Features/Profile/ProfileViewModelBase.cs

```diff
@@ -102,45 +99,122 @@ protected async Task _initialise()
 
     protected async Task LoadProfileSections()
     {
-        if (!_loadingProfileSectionsSemaphore.Wait(0))
+        if (!await _loadingProfileSectionsSemaphore.WaitAsync(0))
             return;
 
         try
         {
             IsLoading = true;
-            var profileTask = _userService.GetUserAsync(UserId);
-            var socialMediaTask = LoadSocialMedia();
-        
-            await Task.WhenAll(profileTask, socialMediaTask);
-
-            var profile = profileTask.Result;
-
-            ProfilePic = profile.ProfilePic ?? "v2sophie";
-            Name = profile.FullName ?? string.Empty;
-            Rank = profile.Rank;
-            Points = profile.Points;
-            Balance = profile.Balance;
-            IsStaff = profile.IsStaff;
-            UserEmail = profile.Email ?? string.Empty;
-            Title = GetTitle();
-        
-            await UpdateSkillsSectionIfRequired();
-            UpdateLastSeenSection(profile.Achievements);
-            UpdateRecentActivitySection(profile.Achievements, profile.Rewards);
+
+            string cacheKey = $"profile_{UserId}";
+
+            await _fileCacheService.FetchAndRefresh(
+                cacheKey,
+                FetchProfileData,
+                OnProfileDataReceived,
+                this
```

Comment:
Ideally, this should be a stateful value, like the date of request or something like that.

```cs
_lastFetchDate = DateTime.UtcNow;

await _fileCacheService.FetchAndRefresh(
    cacheKey,
    FetchProfileData,
    OnProfileDataReceived,
    _lastFetchDate);


// OnProfileDataReceived
if (tag != _lastFetchDate)
{
    // A new request was made while the previous one was in progress. Ignore it.
    return;
}